Add reading log option for 'Stopped Reading'

Yashs911 opened the pull request:
<!-- What issue does this PR close? -->
Closes #2016; Closes #4989 

<!-- What does this PR achieve? [feature|hotfix|fix|refactor] -->
Added "Stopped Reading" option.
![image](https://user-images.githubusercontent.com/64412143/121534330-c05bbf80-ca1e-11eb-8268-780fbed81e45.png)


### Screenshot
<!-- If this PR touches UI, please post evidence (screenshots) of it behaving correctly. -->
![image](https://user-images.githubusercontent.com/64412143/113317646-3364fd80-932d-11eb-886e-d62d0042eef5.png)

### Stakeholders
<!-- @ tag stakeholders of this bug -->
@mekarpeles

jamesachamp commented:
The `ForeignKeyViolation` error is due to the new "Stopped Reading" entry not being present in the database.  After inserting this locally (and fixing the `contact_id` error), I was able to add books to the "Stopped Reading" shelf and view them on the reading log page:

![stopped-reading-log-page](https://user-images.githubusercontent.com/28732543/121446412-706af300-c961-11eb-839d-3fa604f5cfbf.png)

In order to insert the new row locally, you will have to attach to the db container and add it there. To do so, first run `docker-compose exec db bash`.  Then, open the Postgres interactive terminal as user openlibrary:
`psql -U openlibrary`
Connect to the database by entering `\c`.
Finally, execute your new insert statement.

Instructions for fixing the `contact_id` error can be found [here](https://github.com/internetarchive/openlibrary/issues/4989#issuecomment-852560508).

jamesachamp commented on openlibrary/templates/account/books.html:
```diff
@@ -29,6 +29,9 @@
   $ og_title = _("Books %(username)s has read", username=userDisplayName)
   $ og_description = _("%(username)s has read %(total)d books. Join %(username)s on OpenLibrary.org and tell the world about the books that you care about.", username=userDisplayName, total=total_items)
+$elif key == 'stopped-reading':
+  $ og_title = _("Books %(username)s have stopped reading", username=userDisplayName)
+  $ og_description = _("%(username)s have stopped reading %(total)d books.", username=userDisplayName, total=total_items)
```
```suggestion
  $ og_title = _("Books %(username)s has stopped reading", username=userDisplayName)
  $ og_description = _("%(username)s has stopped reading %(total)d books.", username=userDisplayName, total=total_items)
```
Small grammar correction.

jamesachamp commented on openlibrary/i18n/messages.pot:
```diff
@@ -786,6 +786,7 @@ msgstr ""
 msgid ""
 "This user has chosen to publicly share the books they are <a "
 "href=\"%(username)s/books/currently-reading\">currently reading</a>, <a "
+"href=\"%(username)s/books/already-read\">have stopped reading</a>, <a "
 "href=\"%(username)s/books/already-read\">have already read</a>, and <a "
 "href=\"%(username)s/books/want-to-read\">want to read</a>!"
```
Ensure that this string is updated in the `.po` files as well.

Yashs911 commented:
@jamesachamp I added the row, and for #4989 I added the contact_id, but it showed an error for contribution_status, so I removed it from the dictionary.
The python tests are failing due to flake8 errors, but I don't think we need to break that line.

jamesachamp commented on openlibrary/core/civicrm.py:
```diff
@@ -80,6 +80,5 @@ def get_sponsorships_by_contact_id(contact_id=None, isbn=None):
         "total_amount": t.pop("total_amount"),
-        "contact_id": t.pop("contact_id"),
-        "contribution_status": t.pop("contribution_status")
+        "contact_id": "270430"
```
This is not the correct place to make the change.  There is a method named `fake_civi` in upstream/account.py that mocks API responses for local environments.  That is what needs to be modified.

https://github.com/internetarchive/openlibrary/blob/76f28b83eec379945fff6e2d66a6138adfad180c/openlibrary/plugins/upstream/account.py#L842-L861

jamesachamp replied:
When you update this, `Completed` should be a good value for `contribution_status`.

Yashs911 replied:
I have made the changes.

jamesachamp reviewed (changes requested):
Go ahead and correct the flake8 errors as well.  I'm pretty sure that our CI will prevent deployment if there are failing tests.  Not to mention that code which adheres to a standardized style tends to be easier to maintain.

jamesachamp reviewed (approved):
Thanks @Yashs911!  This looks good to me.

jamesachamp commented:
**Important:** The following DML must be executed before this code is deployed:

`INSERT INTO bookshelves (name, description) VALUES ('Stopped Reading', 'A list of books I have stopped reading');`

@mekarpeles @cdrini

mekarpeles commented:
Staff has been struggling to decide whether adding a new "Stopped Reading" shelf ultimately makes sense. For now, I think we are going to close and keep on standby in case there's interest from the community. This PR is very helpful for historical reference as it shows how we may extend our reading log.
